samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,271 @@
+        "   {%- for action in action_list %}\n"
+        "     # - {{action}}\n"
+        "   {%- endfor %}\n"
+        "  Resource: {{arn}}\n"
+    )
+
+    for arn_regex, action_list in default_permissions_map.items():
+        if re.search(arn_regex, resource_arn) is not None:
+            LOG.info("Matched ARN `%s` to IAM actions %s", resource_arn, action_list)
+
+            # APIGW API IAM Statements:
+            # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+            if arn_regex in (APIGW_API_REGEX, APIGW_RESTAPI_REGEX):
+                apiId = _extract_api_id_from_arn(resource_arn)
+                execute_api_arn = (
+                    f"!Sub arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/*/GET/*"
+                )
+                return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
+                    action_list=action_list, arn=execute_api_arn
+                )
+            else:
+                return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
+                    action_list=action_list, arn=resource_arn
+                )
+
+    # If the supplied ARN is for a resource without any default actions supported
+    LOG.info("Found no match for ARN `%s` to any IAM actions.", resource_arn)
+    return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
+        action_list=[], arn=resource_arn
+    )
+
+    return default_permissions_map
+
+
+def _query_tagging_api(tag_filters: dict, boto_client_provider: BotoProviderType) -> Union[dict, None]:
+    """
+    Queries the Tagging API to retrieve the ARNs of every resource with the given tags.
+
+    NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#get-resources
+
+    Parameters
+    ----------
+    tag_filters : dict
+        The tag filters to restrict output to only those resources with the given tags.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#options
+
+    boto_client_provider : BotoProviderType
+        Provides a boto3 client in order to query tagging API
+
+    Returns
+    -------
+    dict
+        A ResourceTagMappingList, which contains a list of resource ARNs and tags associated with each.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#output
+
+    Raises
+    ------
+    botocore.ClientError
+        If the `get_resources` call fails.
+
+        See # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#parsing-error-responses-and-catching-exceptions-from-aws-services
+    """
+    return boto_client_provider("resourcegroupstaggingapi").get_resources(TagFilters=tag_filters)[
+        "ResourceTagMappingList"
+    ]
```

Comment:
I would recommend using `get` and probably with default values so that it will be easier to handle some of the edge cases.

If you use `["KeyName"]` directly, it will raise an error if the key is not there. If you use `get`, it will return `None` if the key is not there. You can also use `get('KeyName', DefaultValue)` as well for dict or list types.

```suggestion
    return boto_client_provider("resourcegroupstaggingapi").get_resources(TagFilters=tag_filters).get("ResourceTagMappingList", [])
```

Reply:
We probably would like to throw an exception if the `ResourceTagMappingList` is not present in the response, because that means the query delivered an unexpected result.

Still use `.get("ResourceTagMappingList")` and if the response is `None`, throw an exception